wpilibNewCommands/src/generate/main/java/edu/wpi/first/wpilibj2/command/button/commandhid.java.jinja

```diff
@@ -0,0 +1,141 @@
+   * @return a Trigger instance that is true when the {{ trigger.DocName }}'s axis exceeds the provided
+   *     threshold, attached to the given event loop
+   */
+  public Trigger {{ trigger.EventMethodName }}(double threshold, EventLoop loop) {
+    return m_hid.{{ trigger.EventMethodName }}(threshold, loop).castTo(Trigger::new);
+  }
+
+  /**
+   * Constructs a Trigger instance around the axis value of the {{ trigger.DocName }}. The returned
+   * trigger will be true when the axis value is greater than {@code threshold}.
+   *
+   * @param threshold the minimum axis value for the returned {@link Trigger} to be true. This value
+   *     should be in the range [0, 1] where 0 is the unpressed state of the axis.
+   * @return a Trigger instance that is true when the {{ trigger.DocName }}'s axis exceeds the provided
+   *     threshold, attached to the {@link CommandScheduler#getDefaultButtonLoop() default scheduler
+   *     button loop}.
+   */
+  public Trigger {{ trigger.EventMethodName }}(double threshold) {
+    return {{ trigger.EventMethodName }}(threshold, CommandScheduler.getInstance().getDefaultButtonLoop());
+  }
+{% endfor -%}
+{% for stick in axes if stick.type == "stick" %}
+  /**
+   * Get the {{ stick.NameParts[1] }} axis value of {{ stick.NameParts[0] }} side of the controller.
+   *
+   * @return The axis value.
+   */
+  public double get{{ stick.NameParts|map("capitalize")|join }}() {
+    return m_hid.get{{ stick.NameParts|map("capitalize")|join }}();
+  }
+{% endfor -%}
+{% for trigger in axes if trigger.type == "trigger" %}
+  /**
+   * Get the {{ trigger.DocName }} axis value of the controller. Note that this axis is bound to the
+   * range of [0, 1] as opposed to the usual [-1, 1].
+   *
+   * @return The axis value.
+   */
+  public double get{{ trigger.HumanName }}Axis() {
+    return m_hid.get{{ trigger.HumanName }}Axis();
+  }
+{% endfor -%}
+{% if ConsoleName == "Xbox" %}
+  /**
+   * Constructs a Trigger instance around the axis value of the left trigger. The returned trigger
+   * will be true when the axis value is greater than 0.5.
+   *
+   * @return a Trigger instance that is true when the left trigger's axis exceeds 0.5, attached to
+   *     the {@link CommandScheduler#getDefaultButtonLoop() default scheduler button loop}.
+   * @deprecated Use lef1tTrigger(0.5) instead
+   */
+  @Deprecated(since = "2024", forRemoval = true)
+  public Trigger leftTrigger() {
+    return leftTrigger(0.5);
+  }
+
+  /**
+   * Constructs a Trigger instance around the axis value of the right trigger. The returned trigger
+   * will be true when the axis value is greater than 0.5.
+   *
+   * @return a Trigger instance that is true when the right trigger's axis exceeds 0.5, attached to
+   *     the {@link CommandScheduler#getDefaultButtonLoop() default scheduler button loop}.
+   * @deprecated Use rightTrigger(0.5) instead
+   */
+  @Deprecated(since = "2024", forRemoval = true)
+  public Trigger rightTrigger() {
```

Comment:
Why are these deprecated?

Reply:
Triggers on Xbox controllers are registered as axes only. On a PlayStation controller, they are registered as both buttons and axes, which creates a naming conflict, so I deprecated the no-arg axis functions and just have teams put in a threshold. Alternatively, I can modify the JSON and template to skip the creation of these axis methods.